crates/ruma-events/src/image_pack.rs

```diff
@@ -0,0 +1,103 @@
+//! Types for image packs in Matrix ([MSC2545]).
+//!
+//! [MSC2545]: https://github.com/matrix-org/matrix-spec-proposals/pull/2545
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use ruma_common::{serde::StringEnum, OwnedMxcUri, OwnedRoomId};
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use crate::{room::ImageInfo, PrivOwnedStr};
+
+/// The content of an `im.ponies.room_emotes` event.
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "im.ponies.room_emotes", kind = State, state_key_type = String)]
+pub struct RoomImagePackEventContent {
+    /// A list of images avaliable.
+    pub images: BTreeMap<String, PackImage>,
+    /// Image pack info.
+    pub pack: PackInfo,
+}
+
+/// The content of an `im.ponies.user_emotes` event.
+#[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "im.ponies.room_emotes", kind = GlobalAccountData)]
+pub struct AccountImagePackEventContent {
+    /// A list of images avaliable.
+    pub images: BTreeMap<String, PackImage>,
+    /// Image pack info.
+    pub pack: PackInfo,
+}
+
+/// An image object in a image pack.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct PackImage {
```

Comment:
This should be non-exhaustive, i.e. have the `#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]` attribute.